### DISCUSSION TITLE
s4m error

### DISCUSSION THREAD

**Discussion by ddm0542** [iainctduncan/scheme-for-max#140] [General]
Hello Iain; I just installed S4M 0.4 on Max 8.3.3 for Windows. It seems to be working ok but when I launch the help file from the package manager the following weird error prints out on the max console:

udpreceiver: binding to port 7000
s4m: s4m Error loading H:/Max/resources/packages/Scheme-For-Max/extras/s4m.scm: 
;cadddr argument, #f, is boolean but should be a pair
;    (let ((returnfn (cadddr (assoc...

But then the max console shows that s4m is initialized and it prints out Hello World from S7 Scheme etc.  As I say the package seems to be working ok but I just wanted to know if this error can be ignored.

Thanks

**Comment by iainctduncan** [comment]
Hi Frank, thanks for the report. I will test this out again on Windows, it's possible one of the instances in the help file is loading funny. Couple of questions:

Does this happen consistently?
What happens if you just put an s4m object in a blank patcher?

In general, if on creation of an s4m object you see the "s4m initialized" message, you should be good. But I would like to figure out why you are getting a funny load.

**Comment by ddm0542** [comment]
Yes it happens every time I launch an instance of the help file.

If I put like an s4m.repl object in a blank patcher there are no errors. So
I guess everything is ok.

**Reply by iainctduncan** [reply]
Hi Frank, it should be an "s4m" object, not the "s4m.repl". Unfortunately C74 (I think?) changed how auto complete works and now it wants to auto complete to s4m.repl all the time, which is a pain. I still need to figure out what to do about that.

**Comment by ddm0542** [comment]
Ha ha yes it did that and I just went with it. So I changed it to an s4m
object with an argument scm file to load, that posted hello world msg, and
there were no error messages on max console
and it posted.

So it looks ok.

Frank

**Reply by iainctduncan** [reply]
Thanks Frank. I will try to figure out why those error messages were happening anyway! If you have further questions, post on the discourse forum and I'll be happy to help!

https://schemeformax.discourse.group/
